### ISSUE TITLE
Make documentation TOC internal link by :CUSTOM_ID:

### ISSUE THREAD

**Issue by louy2** [syl20bnr/spacemacs#2368]
Just found how org internal link works. Let's make the TOC work inside as well!

#1568 #2353 
https://github.com/snosov1/toc-org/issues/1

**Comment by TheBB** [comment]
I also really want links to work internally in Emacs, but it would be a shame to have `:PROPERTIES:` under every heading. We could write a small minor mode to hide them completely, which is only triggered when opening via `spacemacs/view-org-file`.

**Comment by louy2** [comment]
Save for reference: https://stackoverflow.com/questions/17478260/completely-hide-the-properties-drawer-in-org-mode
It needs to match preceding stuff though.

**Comment by louy2** [comment]
It would now be displayed as such:
![2015-07-29_19-18-05](https://cloud.githubusercontent.com/assets/2263580/8972297/b77b7f28-3626-11e5-8ae8-85f32482f8db.png)

I am considering not putting it into `toc-org` though, because it is more general than that. Maybe into extension?

**Comment by TheBB** [comment]
I don't know what happend, but internal links work for me now.

**Comment by louy2** [comment]
`toc-org` now changes `org-link-translation-function` to make internal links work. Forgot to close this.

**Comment by TheBB** [comment]
Did we have a regression here? After last update it seems not to work any more.

**Comment by snosov1** [comment]
Hey, @TheBB !

What exactly stopped working for you? I've tried several files in `spacemacs/doc` (several links in each). It seems to work fine.

In case you'll come up with a reproducible failing scenario, please, file a bug in https://github.com/snosov1/toc-org

**Comment by TheBB** [comment]
Hey @snosov1. I just updated to latest develop with a fresh install of the dependent packages, and no internal links work. I'm using `C-c C-o` to open them (`org-open-at-point`). I can confirm that the value of `org-link-translation-function` is indeed `toc-org-unhrefify`, but this function is never called. I put a `message` in there and eval'd it to be sure.

I tried this both in the Spacemacs `DOCUMENTATION.org` and a fresh Org buffer with contents

```
* Contents :TOC@4:

* Heading
```

After saving, a link to `#heading` appears, but when following it, no dice. The exact error message is "No match for custom ID: heading".
